src/Servers/HttpSys/src/RequestProcessing/RequestStream.cs

```diff
@@ -104,7 +104,7 @@ private static void ValidateReadBuffer(byte[] buffer, int offset, int size)
         {
             throw new ArgumentOutOfRangeException(nameof(offset), offset, string.Empty);
         }
-        if (size <= 0 || size > buffer.Length - offset)
+        if (size < 0 || size > buffer.Length - offset)
```

Comment:
```suggestion
        if ((uint)size > (uint)(buffer.Length - offset))
```
Saves one comparison. Same could be done in L103 (GH doesn't allow me to comment there).